src/ingestors/zora/index.ts

```diff
@@ -0,0 +1,92 @@
+import { DATE_DISTANT_FUTURE, MintTemplateBuilder } from '../../lib/builder/mint-template-builder';
+import { zoraMintAbi } from './abi';
+import {  getZoraMintPriceInEth,getZoraContractMetadata } from './onchain-zora';
+import { fetchCreatorProfile, urlForValidZoraPage,zoraOnchainIdDataFromUrl } from './offchain-metadata';
+
+export class ZoraIngestor implements MintIngestor {
+
+  configuration = {
+    supportsContractIsExpensive: true,
+  };
+
+  async supportsUrl(_resources: MintIngestorResources, url: string): Promise<boolean> {
+    const urlObject = new URL(url);
+    return urlObject.hostname === 'zora.co' && urlObject.pathname.startsWith('/collect/base');
+  }
+  
+  async supportsContract(resources: MintIngestorResources, contract: MintContractOptions): Promise<boolean> {
+    const { contractAddress,tokenId } = contract;
+    if (!contractAddress || !tokenId) {
+      return false;
+    }
+    const url = await urlForValidZoraPage( contractAddress,tokenId, resources.fetcher);
+    return !!url;
+  }
+
+  async createMintTemplateForUrl(resources: MintIngestorResources, url: string): Promise<MintTemplate> {
+    const isCompatible = await this.supportsUrl(resources, url);
+    if (!isCompatible) {
+      throw new MintIngestorError(MintIngestionErrorName.IncompatibleUrl, 'Incompatible URL');
+    }
+
+    const { chainId,tokenId, contractAddress } = await zoraOnchainIdDataFromUrl(url, resources.fetcher);
+    if (!chainId || !contractAddress || !tokenId) {
+      throw new MintIngestorError(MintIngestionErrorName.MissingRequiredData, 'Missing required data');
+    }
+    return this.createMintForContract(resources, {chainId, contractAddress, tokenId, url });
+  }
+
+  async createMintForContract(resources: MintIngestorResources, contract: MintContractOptions): Promise<MintTemplate> {
+    const { chainId, contractAddress,tokenId } = contract;
+    if (!chainId || !contractAddress || !tokenId) {
+      throw new MintIngestorError(MintIngestionErrorName.MissingRequiredData, 'Missing required data');
+    }
+
+    const mintBuilder = new MintTemplateBuilder()
+      .setMintInstructionType(MintInstructionType.EVM_MINT)
+      .setPartnerName('Zora')
+      .setMintOutputContract({chainId, address: contractAddress })
+      .setMarketingUrl(contract.url || `https://zora.co/collect/${'base'}:${contractAddress}${tokenId ? `/${tokenId}` : ''}`);
+      
+    const { name, description, imageUrl, startAt, creatorAddress } = await getZoraContractMetadata(Number(tokenId), contractAddress);
+    mintBuilder.setName(name)
+      .setDescription(description)
+      .setFeaturedImageUrl(imageUrl);
+
+    const totalPriceWei = await getZoraMintPriceInEth(chainId, contractAddress, resources.alchemy);
+    mintBuilder.setMintInstructions({
+      chainId,
+      contractAddress,
+      contractMethod: 'mintWithRewards',
+      contractParams: '[address,1,1,bytes,address]',
```

Comment:
I just ran the tests locally and this won't work since the contract params are incorrect:

* Param 1: This is the minter contract. Most on base use `0x04e2516a2c207e84a1839755675dfd8ef6302f0a` but may be worth looking at when that's _not_ true
* Param 2: Is tokenId - this should be your tokenId value above
* Param 3 is quantity. That's fine to hardcode as 1 for now
* Param 4 is their encoded params. Fortunately, you can actually just pass the macro `encodedAddress` here
* Param 5 is the referrer - for mobile minting, if we have it, set the referral address to the wallet address of the creator. Otherwise use `0xBcaf781ddD26054AF684fb8785c8410EDddCC98a`

```ts
const ZORA_FIXED_PRICE_MINTER_ADDRESS = '0x04e2516a2c207e84a1839755675dfd8ef6302f0a';

I think this would look something like this... (although you should merge main into your branch so that when you run the tests it runs an onchain simulation to check!)
...

let rewardsAddress = creatorAddress || '0xBcaf781ddD26054AF684fb8785c8410EDddCC98a';
let minter = ZORA_FIXED_PRICE_MINTER_ADDRESS;
let params= `["${minter}", ${tokenId || 1 }, 1, encodedAddress, "rewardsAddress"]`
```

Hopefully this helps.

This is all documented in the ABI you included above:

```
{
              'internalType': 'contract IMinter1155',
              'name': 'minter',
              'type': 'address'
          },
          {
              'internalType': 'uint256',
              'name': 'tokenId',
              'type': 'uint256'
          },
          {
              'internalType': 'uint256',
              'name': 'quantity',
              'type': 'uint256'
          },
          {
              'internalType': 'bytes',
              'name': 'minterArguments',
              'type': 'bytes'
          },
          {
              'internalType': 'address',
              'name': 'mintReferral',
              'type': 'address'
          }
```